### PR TITLE
*: record cop scan rows in explain analyze

### PR DESCRIPTION
**Pull request by lamxTyler**
### What problem does this PR solve? 

The cop scan rows were not recorded in explain analyze.

### What is changed and how it works?

Use the stats feedback info to collect the cop scan rows, and record it when the distsql exectuor exits.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Side effects

 - None

Related changes

 - None

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/8423)

### CODE REVIEW

File: executor/distsql.go [hunk 1]
```diff
@@ -521,6 +525,10 @@ func (e *IndexLookUpExecutor) Close() error {
 	e.finished = nil
 	e.memTracker.Detach()
 	e.memTracker = nil
+	if e.runtimeStats != nil {
+		copStats := e.ctx.GetSessionVars().StmtCtx.RuntimeStatsColl.Get(e.idxPlans[0].ExplainID())
```
Comment (zz-jason) [inline executor/distsql.go]: Uh, so we can not record the table scan side of the index lookup reader?

File: executor/builder.go [hunk 2]
```diff
@@ -1679,7 +1679,10 @@ func buildNoRangeTableReader(b *executorBuilder, v *plannercore.PhysicalTableRea
 	} else {
 		e.feedback = statistics.NewQueryFeedback(e.physicalTableID, ts.Hist, int64(ts.StatsCount()), ts.Desc)
 	}
-	collect := e.feedback.CollectFeedback(len(ts.Ranges))
+	collect := (b.ctx.GetSessionVars().StmtCtx.RuntimeStatsColl != nil) || e.feedback.CollectFeedback(len(ts.Ranges))
+	if !collect {
+		e.feedback.Invalidate()
```
[the remark this answers is not in the document]
Comment (lamxTyler) [reply executor/builder.go]: No. The invalidation operation was in `CollectFeedback`, now I move it out.

File: executor/distsql.go (hunk 1 above)
Comment (lamxTyler) [reply executor/distsql.go]: Because the table side scan rows are not so important.

Comment (zz-jason) [reply executor/distsql.go]: Uh, I think it sometimes can be useful, for example:
```
index lookup reader
|-> index scan
|
+-> selection
	|
	+-> table scan
```

with the row count of index scan and selection, we can check the selectivity estimated for selection.

Comment (eurekaka) [reply executor/distsql.go]: For this case, we can know the row count of `table scan` and `index lookup reader` in the output of `explain analyze`, so we can get the selectivity of this `selection`?

File: executor/distsql.go [hunk 3]
```diff
@@ -237,9 +237,13 @@ type IndexReaderExecutor struct {
 
 // Close clears all resources hold by current object.
 func (e *IndexReaderExecutor) Close() error {
-	e.ctx.StoreQueryFeedback(e.feedback)
 	err := e.result.Close()
 	e.result = nil
+	if e.runtimeStats != nil {
+		copStats := e.ctx.GetSessionVars().StmtCtx.RuntimeStatsColl.Get(e.plans[0].ExplainID())
```
[the remark this answers is not in the document]
Comment (lamxTyler) [reply executor/distsql.go]: Yes.

File: executor/distsql.go (hunk 1 above)
Comment (eurekaka) [reply executor/distsql.go]: Oh, row count of `table scan` is added to address this comment, got it.

**Pull request merged**